src/JsonApi/EventListener/TransformSortingParametersListener.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\JsonApi\EventListener;
+
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * Converts pagination parameters from JSON API recommended convention to
+ * api-platform convention.
+ *
+ * @see http://jsonapi.org/format/#fetching-sorting
+ * @see https://api-platform.com/docs/core/filters#order-filter
+ *
+ * @author Héctor Hurtarte <hectorh30@gmail.com>
+ */
+final class TransformSortingParametersListener
```

Comment:
It feels like those `parametersListener` have a lot of code duplication. I don't know if it's okay like this or if we should base those on an abstract class. WDYT?